src/Parallel/OutputInbox.hpp

```diff
@@ -0,0 +1,46 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "Utilities/TypeTraits/CreateIsCallable.hpp"
+
+namespace Parallel {
+namespace detail {
+CREATE_IS_CALLABLE(output_inbox)
+CREATE_IS_CALLABLE_V(output_inbox)
+}  // namespace detail
+
+/*!
+ * \brief Returns a string of the contents of the inbox.
+ *
+ * Calls a static `output_inbox` member function of the `InboxTag`. If the inbox
+ * tag doesn't have the `output_inbox` function, then this function can't be
+ * called with that inbox (a `static_assert` will fail).
+ *
+ * \note It's up to the individual inbox what to include in the string. Data may
+ * or may not be included
+ *
+ * \tparam InboxTag Inbox tag with an `output_inbox` member function
+ * \tparam InboxTypes Types of all inboxes (deduced)
+ * \param inboxes All inboxes
+ * \param indent_pad Number of empty spaces to pad the message with incase
+ * indenting is needed.
+ * \return std::string The contents of the inbox
+ */
+template <typename InboxTag, typename... InboxTypes>
+std::string output_inbox(const tuples::TaggedTuple<InboxTypes...>& inboxes,
+                         const size_t indent_pad) {
+  static_assert(tmpl::list_contains_v<tmpl::list<InboxTypes...>, InboxTag>);
+  static_assert(detail::is_output_inbox_callable_v<
+                    InboxTag, const typename InboxTag::type&, const size_t>,
+                "To output an inbox, there must be a static 'output_inbox' "
+                "function defined in the inbox tag.");
```

Comment:
I generally dislike this sort of check, because I find that it makes error messages less useful.  Here you'll get a generic "there must be a static `output_inbox' function", whereas without this the compiler can specifically say the function doesn't exist or it's not static or it can't deduce a template parameter or the argument type is wrong or whatever.